packages/core/snapshotcreator/snapshotcreator.go

```diff
@@ -114,7 +116,7 @@ func CreateSnapshot(opts ...options.Option[Options]) error {
 }
 
 func (m *Options) attest(engineInstance *engine.Engine, nodePublicKey ed25519.PublicKey) error {
-	if _, err := engineInstance.NotarizationManager.Attestations.Add(&notarization.Attestation{
+	if _, err := engineInstance.Notarization.Attestations().(*slotnotarization.Attestations).Add(&notarization.Attestation{
```

Comment:
Why not add these to the interface? To keep it lean as it's only needed here and in some tests?

Reply:
Yes, see other comment above.